libass/ass.c

```diff
@@ -264,14 +264,24 @@ static long long string2timecode(ASS_Library *library, char *p)
 
 #define STRVAL(name) \
     } else if (ass_strcasecmp(tname, #name) == 0) { \
-        if (target->name != NULL) free(target->name); \
-        target->name = strdup(token);
+        char* new_str = strdup(token); \
+        if (new_str) { \
+            free(target->name); \
+            target->name = new_str; \
+        } else { \
+            ass_nomem_log(track->library, true, "Failed to strdup field '%s'", #name); \
```

Comment:
I suppose this could use string literal concatenation: `"Failed to strdup field '" #name "'"`. But either way is fine by me.

Reply:
Since `name` will be a valid C variable name it can't contain printf-sequences and literal concatenation should thus be safe. *(probably also a bit better in regard to memory usage)*
Changed it.